client/transaction-pool/src/lib.rs

```diff
@@ -127,28 +129,40 @@ impl<PoolApi, Block> TransactionPool for BasicPool<PoolApi, Block>
 	fn submit_at(
 		&self,
 		at: &BlockId<Self::Block>,
-		xts: impl IntoIterator<Item=TransactionFor<Self>> + 'static,
-	) -> Box<dyn Future<Output=Result<Vec<Result<TxHash<Self>, Self::Error>>, Self::Error>> + Send + Unpin> {
-		Box::new(self.pool.submit_at(at, xts, false))
+		xts: Vec<TransactionFor<Self>>,
+	) -> PoolResult<Vec<Result<TxHash<Self>, Self::Error>>> {
+		let pool = self.pool.clone();
```

Comment:
I removed lifetimes, but this seems to be unavoidable

Basically, due to how BasicPool is used elsewhere (rpc/etc), it has to carry shared pool reference with it